src/iuse.cpp

```diff
@@ -1102,6 +1102,58 @@ int iuse::purify_iv( player *p, item *it, bool, const tripoint & )
     return it->type->charges_to_use();
 }
 
+int iuse::purify_smart( player *p, item *it, bool, const tripoint & )
+{
+    mutagen_attempt checks = mutagen_common_checks( *p, *it, false,
+                             pgettext( "memorial_male", "Injected smart purifier." ), pgettext( "memorial_female",
+                                     "Injected smart purifier." ) );
+    if( !checks.allowed ) {
+        return checks.charges_used;
+    }
+
+    std::vector<trait_id> valid; // Which flags the player has
+    std::vector<std::string> valid_names; // Which flags the player has
```

Comment:
This is "which flags the player has" (according to the comment), but so is `valid` above. What's their difference, why are there two variables that - according to the comment - do / are the same?